src/test/regress/expected/subquery_in_targetlist.out

```diff
@@ -0,0 +1,275 @@
+CREATE SCHEMA subquery_in_targetlist;
+SET search_path TO subquery_in_targetlist, public;
+-- simple empty target list
+SELECT event_type, (SELECT 1 + 3 + e.value_2)
+FROM events_table e
+ORDER BY 1,2 LIMIT 1;
+ event_type | ?column?
+---------------------------------------------------------------------
+          0 |        4
+(1 row)
+
+-- simple reference table sublink
+SELECT event_type, (SELECT user_id FROM users_reference_table WHERE user_id = 1 AND value_1 = 1)
+FROM events_table e
+ORDER BY 1,2 LIMIT 1;
+ event_type | user_id
+---------------------------------------------------------------------
+          0 |       1
+(1 row)
+
+-- duplicate vars
+SELECT event_type, (SELECT e.value_2 FROM users_reference_table WHERE user_id = 1 AND value_1 = 1), (SELECT e.value_2)
+FROM events_table e
+ORDER BY 1,2 LIMIT 1;
+ event_type | value_2 | value_2
+---------------------------------------------------------------------
+          0 |       0 |       0
+(1 row)
+
+-- correlated subquery with aggregate
+SELECT event_type, (SELECT max(time) FROM users_table WHERE user_id = e.user_id)
+FROM events_table e
+ORDER BY 1,2 LIMIT 1;
+ERROR:  cannot push down this subquery
+DETAIL:  Aggregates without group by are currently unsupported when a subquery references a column from another query
+-- correlated subquery wtth limit
+SELECT event_type, (SELECT time FROM users_table WHERE user_id = e.user_id ORDER BY time LIMIT 1)
+FROM events_table e
+ORDER BY 1,2 LIMIT 1;
+ERROR:  cannot push down this subquery
+DETAIL:  Limit in subquery is currently unsupported when a subquery references a column from another query
+-- correlated subquery with group by distribution column
+SELECT event_type, (SELECT max(time) FROM users_table WHERE user_id = e.user_id GROUP BY user_id)
+FROM events_table e
+ORDER BY 1,2 LIMIT 1;
+ event_type |              max
+---------------------------------------------------------------------
+          0 | Thu Nov 23 13:52:54.83829 2017
+(1 row)
+
+-- correlated subquery with group by almost distribution column
```

Comment:
One thing that we can consider doing in the future is to relax the distribution key checks (e.g., GROUP BY dist_key) based on restriction equivalence. We know that there is an equality between `user_id` of the both relations. 

For example:
```SQL
SELECT
  (SELECT sum(user_id)
   FROM users_table
   WHERE user_id = u1.user_id
   GROUP BY user_id)
FROM users_table u1
GROUP BY user_id;
 sum 
-----
 102
 260
 184
 120
  72
  14
(6 rows)

-- same query with sum(u1.user_id) instead of sum(user_id) fails
-- in fact, these are the same queries
 SELECT
  (SELECT sum(u1.user_id)
   FROM users_table
   WHERE user_id = u1.user_id
   GROUP BY user_id)
FROM users_table u1
GROUP BY user_id;
ERROR:  cannot push down subquery on the target list
```

Reply:
Makes sense, though for correlated subqueries we don't need to care which column is used in the GROUP BY.

Reply:
I think it passes? https://github.com/citusdata/citus/pull/4385/files#diff-d0c32107a00ede708fd64b3cbe9ad9e38a2da6ad0aaa8f2b675097232090ca1eR91